scipy/stats/stats.py

```diff
@@ -239,6 +239,79 @@ def _chk2_asarray(a, b, axis):
     return a, b, outaxis
 
 
+def _shape_with_dropped_axis(a, axis):
+    """
+    Given an array `a` and an integer `axis`, return the shape
+    of `a` with the `axis` dimension removed.
+
+    Examples
+    --------
+    >>> a = np.zeros((3, 5, 2))
+    >>> _shape_with_dropped_axis(a, 1)
+    (3, 2)
+    """
+    shp = list(a.shape)
+    try:
+        del shp[axis]
+    except IndexError:
+        raise np.AxisError(axis, a.ndim) from None
+    return tuple(shp)
+
+
+def _broadcast_shapes(shape1, shape2):
```

Comment:
I can imagine that this function has greater applicability than just this one module.  Is this the right location for these three functions?

Reply:
> I can imagine that this function has greater applicability ...

Me too.  In fact, I was surprised to find that it doesn't already exist in numpy.

Having said that, I don't have any other applications for these at the moment, so I think it is OK to leave them here.

Reply:
I know I've seen code like this elsewhere in the codebase. For example, these could immediately be dropped in here I think:

https://github.com/scipy/scipy/blob/master/scipy/signal/spectral.py#L1698-L1705

You don't have to make the substitution in this PR if you don't want to, but I agree with @pvanmulbregt that it makes sense to at least put these functions in `scipy._lib` or so such that the next PR can do it more easily.

I also agree this is something that should probably exist in `numpy`, it seems like a `numpy.broadcast_shapes` would be a natural complement to `numpy.broadcast`.